src/mate/academy/homework11/regularExpresion/Main.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.homework11.regularExpresion;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Main {
+    public static void main(String[] args) {
+        String text = getWord();
+        workWithText(text);
+    }
+
+    static String getWord() {
+        String result = " ";
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+            result = reader.readLine();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    static void workWithText(String text) {
+        Map<String, Integer> maps = new HashMap<>();
+        Pattern pattern = Pattern.compile("\\b\\w+\\b", Pattern.UNICODE_CHARACTER_CLASS);
+        Matcher matcher = pattern.matcher(text);
+
+        while (matcher.find()) {
```

Comment:
imperative code, you should use streams, here you have "what to do" and "how to do", streams allow you to have "what to do" code